pkg/local_object_storage/blobovnicza/delete.go

```diff
@@ -29,7 +29,7 @@ func (p *DeletePrm) SetAddress(addr *addressSDK.Address) {
 // Returns an error of type apistatus.ObjectNotFound if the object to be deleted is not in blobovnicza.
 //
 // Should not be called in read-only configuration.
-func (b *Blobovnicza) Delete(prm *DeletePrm) (*DeleteRes, error) {
+func (b *Blobovnicza) Delete(prm DeletePrm) (DeleteRes, error) {
```

Comment:
Why did you decide to return a value instead of a pointer?

Reply:
what do you mean? what that PR is doing is removing poiners in the whole SE (since it was suggested in #1398 and I really do not know why we need pointers in that cases). so you asking why does that PR exist in general?

Reply:
I understand PR existence, and accepting parameters by value is really good. But what about return values? The common approach we use is returning `(*T, error)`. What are pros to return `(T, error)`? I mean why don't we return just `T` which contains `error`? Maybe I've missed some discussion.

Reply:
> The common approach we use is returning (*T, error).

> i saw some examples in the SE where it already was as `func(prm Prm) (Res, error)` why is it common? to be able to return `nil` res?

> I mean why don't we return just T which contains error

to support common practice in `go`: start handling results only if `err == nil` (and know that operation could be finished with error and do not read docs to the `Res` struct)

> Maybe I've missed some discussion.

no, that was my decision

Reply:
>What are pros to return `(T, error)`?

If we return a pointer, the struct will _definitely_ be allocated on heap (all these functions are big enough not to be inlined). And in case of `T` there is no allocation. It is more beneficial than passing parameters by value, actually, because parameters can still be allocated on stack as they do not outlive the called function.

>I mean why don't we return just T which contains error

This is possible, but would make the code non-idiomatic for no real reason IMO. We should still check for the error.

Reply:
Until the building itself, we do not know exactly how the data will be allocated since this is not visible from the source code. Are there any objective reasons for the change? If it's really a bottleneck, then we need to provide test proof.

We've already discussed this earlier, and previously I voted against `return (*T, error)` for the same GC optimization and suggested to accept result by pointer as a parameter. This will allow to reuse the same instance between several calls, and most definitely avoid heap allocation. But we've decided with @carpawell @alexvanin to not do it like this. At the same time I don't see returning `(T, error)` convenient.

I suggest to see what the profiler shows.

Reply:
1. There is no "heap" in terms Go source code. In your example you make a decision according to the specific compiler. What does it mean "big enough not to be inlined"? How can I get it from the sources?

And again... I am aware and agree that returning a pointer leads rather to worse performance. Therefore, I propose a compromise: allocate a structure to record the results on the calling side and pass a pointer to it in the function.

Reply:
i reorganized that PR: it now only contains `*Prm` -> `Prm` changes

`Res` changes will be added via another PR